src/utils/types.js

```diff
@@ -0,0 +1,10 @@
+const type = {
+  isArray: (value) => value && typeof value === 'object' && value.constructor === Array,
+  isBoolean: (value) => typeof value === 'boolean',
+  isFunction: (value) => typeof value === 'function',
+  isNumber: (value) => typeof value === 'number' && Number.isFinite(value),
+  isObject: (value) => typeof value === 'object' || value.constructor === Object,
```

Comment:
The proper `isObject` logic:

```js
isObject: (value) => typeof value === 'object' && value !== null
```

The logic you have here is wrong for 2 reasons:
1. Any value whose `constructor` is `Object` will have `typeof value === 'object'`, so that extra check is redundant (yes, you can technically take a number and set `num.constructor = Object` but would you call that an object?)
2. `null` will return true for `typeof value === 'object'`. This is one of those weird JS nuances that is leftover from an older time. Regardless, whenever you are checking if `typeof` is `'object'` you must also check for `null`.

Reply:
My source article for the object check was definitely wrong on this one. Thanks for pointing out the issue

Reply:
Actually it was me that was wrong. I must have modified the check by accident at some point. The [article](https://webbjocke.com/javascript-check-data-types/) correctly enforces a check for null. The reason for the constructor check is for objects created from classes